### PR TITLE
Add more questions to Rust at Work and Rust in Education sections

### PR DESCRIPTION
**Pull request by rylev**
This adds a lot of questions to the questions.md file for the Rust at Work and Rust in Education sections. 

Some notes: 
* I added any meta commentary in block quotes. This seems to be a better approach then putting the questions themselves in the block quotes
* A lot of the questions that I added are actually marked as removed. I wanted to make sure we capture why we're removing certain questions.

### CODE REVIEW

File: surveys/2021-annual-survey/questions.md [hunk 1]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+Select one:
+- My company uses Rust for a large portion of production projects.
+- My company uses Rust for a small portion of production projects.
+- My company uses Rust only for non-production projects (e.g., tooling).
+- My company has actively experimented with Rust for use in production projects.
+- My company has actively experimented with Rust for use in non-production projects.
+- My company has seriously discussed but not experimented with using Rust.
+- My company has not seriously considered Rust for any use.
+- I am unsure whether my company has considered using or currently uses Rust.
+- I don't work for a company or my company does not develop software of any kind.
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:254]: These answers seem a little too fine-grained to me, I would probably half the number of options

File: surveys/2021-annual-survey/questions.md [hunk 2]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+
+> **justification**
+>
+> We want to establish how reliant companies are on Rust.
+
+### In what technology domains is Rust used at your company?
+
+Select as many as apply:
+- backend application
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:263]: I would rephrase this, 'backend' is heavily overloaded and (to me at least) overlaps significantly with web application

File: surveys/2021-annual-survey/questions.md [hunk 3]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
-- server networking, desktop application, mobile application, web application, embedded, etc.
+### How could we make Rust more appealing to your company?
+
+Free form:
+
+> **REMOVED**
+>
+> This question is too vague and unlikely to yield any interesting answers.
+
+### How may developers at your company use Rust at work?
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:322]:
```suggestion
### How many developers at your company use Rust at work?
```

File: surveys/2021-annual-survey/questions.md [hunk 4]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+**TODO**: explain why the following questions are not asked
 Is Rust used in the company where you work?
 
-- yes by > 100 devs
-- yes < 100
-- yes < 50
-- yes < 20
-- yes < 10
-- yes < 5
+### To why extent is Rust currently being used by your company?
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:243]:
```suggestion
### To what extent is Rust currently being used by your company?
```

File: surveys/2021-annual-survey/questions.md [hunk 5]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+> where more than 100 use Rust is different than a company with 110 developers. We can gauge the importance
+> of Rust to companies through more direct questions
+
+### Approximately how many total developers does your company employ?
+
+- under 10
+- 11-49
+- 50-99
+- 100-500
+- 100-500
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:344]:
```suggestion
```

File: surveys/2021-annual-survey/questions.md [hunk 6]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+- 500-1,000
+- 1,000-10,000
+- 10,000-100,000
+- Over 100,000
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:348]: I would probably go for "over 10,000" in place of the last two (given the number of companies with more than 10000 devs)

File: surveys/2021-annual-survey/questions.md [hunk 7]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
+> role in professional settings over time, and this is better served by other questions.
+
+### Is your company planning on hiring Rust developers in the next year?
+
+Select one:
+- Yes
+- No
+- I don't know
+
+> **REMOVED**
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:377]: I'm not sure about this one, at first I thought we should remove it, but on reflection I think it is quite a good question to assess hiring sentiment - although there is intrinsic uncertainty, it is easy to answer and forward looking.

File: surveys/2021-annual-survey/questions.md [hunk 8]
```diff
@@ -219,70 +219,213 @@ TODO ask about blockers for participation?
...
 - Other course which uses Rust
 - Individual project
 - Group project
 - Other activity
 
-Is Rust mandated for your course or activity, or did you choose it yourself?
+### Is Rust mandated for your course or activity, or did you choose it yourself?
 
 - Compulsory
 - Optional (but suggested in some way)
```
Comment (nrc) [inline surveys/2021-annual-survey/questions.md:431]:
```suggestion
- Optional
```

It may or may not be suggested, e.g., an individual project might be allowed to be done in any language and the student chose Rust with no suggestion.

**Review by nrc — COMMENTED** [review]
Looks good! I think it would be useful for the non-removed questions to mark if they were in previous years' surveys or not.

File: surveys/2021-annual-survey/questions.md (hunk 1 above)
Comment (rylev) [reply surveys/2021-annual-survey/questions.md:254]: Can you explain more on which options you find to be redundant? While there are quite a few options here, I think they are all useful to differentiate.
